compiler/rustc_trait_selection/src/traits/structural_match.rs

```diff
@@ -38,22 +38,6 @@ pub fn search_for_structural_match_violation<'tcx>(
         .break_value()
 }
 
-/// This method traverses the structure of `ty`, trying to find any
-/// types that are not allowed to be used in a const generic.
-///
-/// This is either because the type does not implement `StructuralEq`
-/// and `StructuralPartialEq`, or because the type is intentionally
-/// not supported in const generics (such as floats and raw pointers,
-/// which are allowed in match blocks).
-pub fn search_for_adt_const_param_violation<'tcx>(
-    span: Span,
-    tcx: TyCtxt<'tcx>,
-    ty: Ty<'tcx>,
-) -> Option<Ty<'tcx>> {
-    ty.visit_with(&mut Search { tcx, span, seen: FxHashSet::default(), adt_const_param: true })
```

Comment:
with this removed I think you can just remove the `adt_const_param` field from `Search`